### PR TITLE
cilium, docker: builder dependency updates

### PR DESCRIPTION
**Pull request by borkmann**
Base cilium-builder on top of cilium-runtime and therefore also automatically import the same clang/llvm version we have in cilium-runtime. Clean up and shrink cilium-builder dependencies.

clang/llvm split off will be done in #10709 since it requires a rebuild of cilium-runtime.

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]
LGTM, thanks!

**Review by qmonnet — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
Travis CI hit #10690, restarting.

**Review by aanm — CHANGES_REQUESTED** [review]
do not merge

File: Dockerfile.builder [hunk 1]
```diff
@@ -1,11 +1,8 @@
 #
-# Cilium build-time dependencies.
-# Image created from this file is used to build Cilium.
+# Cilium build-time base image (image created from this file is used to build Cilium)
 #
-FROM docker.io/library/ubuntu:18.04
-
+FROM quay.io/cilium/cilium-runtime:2020-03-23
```
Comment (aanm) [inline Dockerfile.builder:4]: This needs to be bumped to `2020-03-26`

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ FROM quay.io/cilium/cilium-envoy:a3385205ad620550b35d3b0b651e40898386e6e3 as cil
 # versions to be built while allowing the new versions to make changes
 # that are not backwards compatible.
 #
-FROM quay.io/cilium/cilium-builder:2020-03-21 as builder
+FROM quay.io/cilium/cilium-builder:2020-03-23 as builder
 LABEL maintainer="maintainer@cilium.io"
 WORKDIR /go/src/github.com/cilium/cilium
 COPY . ./
```
Comment (aanm) [inline Dockerfile:19]: The runtime image needs to be bumped to `2020-03-26`

**Review by aanm — CHANGES_REQUESTED** [review]
Moving the dockerfile around will brake quay builds. I've created a new quay job which accounts for the dockerfile to be `/Dockerfile.runtime`.

**Comment by borkmann** [comment]
Discussed with @aanm offline that we should move all except the official/top-level  `cilium-docker-plugin.Dockerfile`, `cilium-operator.Dockerfile` and `Dockerfile` into contrib/packaging/docker/.

File: Dockerfile.builder [hunk 3]
```diff
@@ -17,43 +14,35 @@ ENV PATH "$GOROOT/bin:$GOPATH/bin:$PATH"
...
+  curl \
+  gcc \
+  git \
+  libc6-dev \
+  libelf-dev \
+  make \
+  rsync \
+  unzip \
+  wget \
+  zip && \
```
Comment (tklauser) [inline Dockerfile.builder:35]: Are these still needed? Would it make sense to document for each of of the installed packages why they're needed?

Comment (borkmann) [reply Dockerfile.builder:35]: Not 100% sure whether some of the above are still needed. I guess I'll find out via trial and error for some of them. ;) I'll document the ones needed, like in cilium-runtime, agree.

**Comment by aanm** [comment]
this PR has been marked as a draft PR since it had a WIP label. Please click in "Ready for review" [below vvv ] once the PR is ready to be reviewed. CI will still run for draft PRs.

**Comment by borkmann** [comment]
Rebased into https://github.com/cilium/cilium/pull/11308 .

**Pull request closed without merging**